644 feature re implement ai assistant with manual history

JoelWiebe opened the pull request:
## Completed

- Added abbreviated IDs and replacement and restoration functions before and after the LLM model calls
- Provide the chat history as context
- Switched from chat to freeform (non-chat) gemini

**Test**

- [x] AI Assistant still operates common commands (e.g., add/remove posts from canvas/buckets and answers questions about the data)
- [x] AI Assistant can recall what it told you in previous messages

Closes #644

JoelWiebe commented:
@markiianbabiak Can you review this one before we start on the Idea Agent?

markiianbabiak reviewed (commented):
[chat_history (8).csv](https://github.com/user-attachments/files/18341163/chat_history.8.csv)
Consider changing the chat history format.

Other than that seems all good to me!

JoelWiebe commented:
> [chat_history (8).csv](https://github.com/user-attachments/files/18341163/chat_history.8.csv) Consider changing the chat history format.
> 
> Other than that seems all good to me!

Thanks! What would you recommend changing it to? Is it the csv filename, filetype, or the layout of the csv that you suggest changing--or something else? We could add a date/timestamp to the chat_history filename that would be a bit more descriptive.

JoelWiebe commented:
Ah, the full log history is being logged for each user entry. I'll fix that.

JoelWiebe commented:
@markiianbabiak I added the timestamp to the filename and fixed the user logs issue for chat history download. Can you double check and approve if good to go?

markiianbabiak commented:
Looks good! Feel free to merge.
